### PR TITLE
BHBC-1021 API built to receive survey creation requests

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- {https://quartech.atlassian.net/browse/BHBC-1021}

## This PR contains the following changes

- new API endpoint for creating surveys

## This PR contains the following types of changes

- [x] New feature (change which adds functionality)

### General

- [x] I have performed a self-review of my own code

### Tests

- [ ] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [x] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [ ] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [x] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

File: api/src/models/survey-create.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { getLogger } from '../utils/logger';
+
+const defaultLog = getLogger('models/survey-create');
+
+/**
+ * Processes all POST /project/{projectId}/survey/create request data.
+ *
+ * @export
+ * @class PostProjectObject
```
Comment (sdevalapurkar) [inline api/src/models/survey-create.ts]:
```suggestion
 * @class PostSurveyObject
```

File: api/src/paths/project/{projectId}/survey/create.ts [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      $ref: '#/components/responses/500'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Creates a new project record.
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/survey/create.ts]:
```suggestion
 * Creates a new survey record.
```

File: api/src/queries/survey/survey-create-queries.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ */
+export const postSurveySQL = (projectId: number, survey: PostSurveyObject): SQLStatement | null => {
+  defaultLog.debug({
+    label: 'postSurveySQL',
+    message: 'params',
+    projectId,
+    survey
+  });
+
+  if (!projectId) {
```
Comment (sdevalapurkar) [inline api/src/queries/survey/survey-create-queries.ts]: just combine the two if checks into one:
```
if (!projectId || !survey) {
    return null;
}
```

File: api/src/queries/survey/survey-create-queries.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+  surveyProprietor: PostSurveyProprietorData
+): SQLStatement | null => {
+  defaultLog.debug({
+    label: 'postSurveyProprietorSQL',
+    message: 'params',
+    surveyId,
+    surveyProprietor
+  });
+
+  if (!surveyId) {
```
Comment (sdevalapurkar) [inline api/src/queries/survey/survey-create-queries.ts]: same thing here to combine checks

**Comment by sdevalapurkar** [comment]
the code smells related to `Remove the unnecessary boolean literal.` should be addressed, the others can be ignored :)

File: api/src/models/survey-create.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  biologist_first_name: string;
+  biologist_last_name: string;
+  foippa_requirements_accepted: boolean;
+  species: string;
+  start_date: string;
+  end_date: string;
+  survey_area_name: string;
+  survey_data_proprietary: boolean;
+  survey_purpose: string;
+  surveyProprietor?: PostSurveyProprietorData;
```
Comment (sdevalapurkar) [inline api/src/models/survey-create.ts]: kind of a nit but should we stick with snake_case here?

call this `survey_proprietor`?

File: api/src/paths/project/{projectId}/survey/create.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+      'req.params': req.params,
+      'req.body': req.body
+    });
+
+    if (!req.params.projectId) {
+      throw new HTTP400('Missing required path param `projectId`');
+    }
+    const connection = getDBConnection(req['keycloak_token']);
+
+    console.log('looking for first_nations_id', req.body);
```
Comment (sdevalapurkar) [inline api/src/paths/project/{projectId}/survey/create.ts]:
```suggestion
```

File: api/src/queries/survey/survey-create-queries.ts [hunk 7]
```diff
@@ -0,0 +1,99 @@
+import { SQL, SQLStatement } from 'sql-template-strings';
+import { getLogger } from '../../utils/logger';
+import { PostSurveyObject, PostSurveyProprietorData } from '../../models/survey-create';
+
+const defaultLog = getLogger('queries/survey/user-queries');
```
Comment (sdevalapurkar) [inline api/src/queries/survey/survey-create-queries.ts]:
```suggestion
const defaultLog = getLogger('queries/survey/survey-create-queries');
```

Comment (anissa-agahchen) [reply api/src/queries/survey/survey-create-queries.ts]: good catch

**Review by sdevalapurkar — APPROVED** [review]
g75csv9111x41

**Pull request merged**